realsense2_camera/include/image_publisher.h

```diff
@@ -22,8 +22,13 @@ class image_publisher
 {
 public:
     virtual void publish( sensor_msgs::msg::Image::UniquePtr image_ptr ) = 0;
+    virtual void publish( std::unique_ptr<cv_bridge::ROSCvMatContainer> container_ptr ){throw std::runtime_error("This publisher does not use type adaptation");};
```

Comment:
This is a pure abstract class and hence should only have pure virtual member functions (no default implementation that should be overwritten). We don't need to store this `has_type_adapter_` boolean. The implementations of this class that do not have the adapter will simple have an empty implementation of this function.